pkg/stdlib/html/input.go

```diff
@@ -2,71 +2,100 @@ package html
 
 import (
 	"context"
+
 	"github.com/MontFerret/ferret/pkg/html/dynamic"
 	"github.com/MontFerret/ferret/pkg/runtime/core"
 	"github.com/MontFerret/ferret/pkg/runtime/values"
 )
 
 /*
- * Sends a value to an underlying input element.
+ * Types a value to an underlying input element.
  * @param source (Document | Element) - Event target.
  * @param valueOrSelector (String) - Selector or a value.
  * @param value (String) - Target value.
+ * @param delay (Int, optional) - Waits delay milliseconds between keystrokes
  * @returns (Boolean) - Returns true if an element was found.
  */
 func Input(_ context.Context, args ...core.Value) (core.Value, error) {
-	err := core.ValidateArgs(args, 2, 3)
+	err := core.ValidateArgs(args, 2, 4)
 
 	if err != nil {
 		return values.None, err
 	}
 
-	// TYPE(el, "foobar")
-	if len(args) == 2 {
+	switch args[0].(type) {
+	case *dynamic.HTMLDocument:
 		arg1 := args[0]
 
-		err := core.ValidateType(arg1, core.HTMLElementType)
+		err = core.ValidateType(arg1, core.HTMLDocumentType)
```

Comment:
You already casted to a target type. This check is redundant.
Let's move you to line 25 and do like this
```go
err = core.ValidateType(arg1, core.HTMLDocumentType, core.HTMLElementType)
```
It will cover both types.